lib/notify.js

```diff
@@ -49,12 +50,20 @@ module.exports = CoreObject.extend({
         .catch(function(error) {
           var errorMessage = serviceKey + ' => ' + error;
 
-          plugin.log(errorMessage, { color: 'red' });
+          if (critical) {
+            return Promise.reject(error);
+          } else {
+            plugin.log(errorMessage, { color: 'red' });
+          }
         });
     } else {
       var warningMessage = 'No request issued! Request options invalid! You have to specify `url`, `headers`, `method` and `body`.';
-      plugin.log(serviceKey+' => '+warningMessage, { color: 'yellow', verbose: true });
 
+      if (critical === true) {
+        return Promise.reject(warningMessage);
+      } else {
+        plugin.log(serviceKey+' => '+warningMessage, { color: 'yellow', verbose: true });
```

Comment:
we don't need the else here. it should be enough if we just put it below the `critical === true`-conditional with the `return Promise.resolve();`-call